sdk/appconfiguration/Azure.Data.AppConfiguration/src/AuthenticationPolicy.cs

```diff
@@ -56,7 +56,7 @@ private async ValueTask ProcessAsync(HttpPipelineMessage message, bool async)
 
             using (var hmac = new HMACSHA256(_secret))
             {
-                Uri uri = message.Request.UriBuilder.Uri;
+                Uri uri = message.Request.Uri.ToUri();
```

Comment:
That doesn't look right: `Uri.ToUri()`? The property already makes it look like a `Uri` given the name, so why would someone think to call `ToUri()`?

Reply:
This usage is strange because it's slightly wrong, the code proceeds to read the same information of `Uri` that was available on `UriBuilder` but I understand you general concern. I think we would have very few places calling `ToUri()` and a LOT accessing `request.Uri` so if we need to make one thing looks nice I would rather improve the later.